### ISSUE TITLE
Migrate PythonTurtle to wxgtk4.0

### ISSUE THREAD

**Issue by bittner** [cool-RR/PythonTurtle#99]
On recent GNU/Linux distributions wxgtk3.0 seems to be gradually phased out in favor of wxgtk4.0. Hence, PythonTurtle should be migrated to wxgtk4.0 to avoid being unmaintainable one day due to unavailable dependencies.

Example of available packages on Ubuntu 18.04 LTS Bionic:
```bash
$ apt-cache search wxgtk
libwxgtk-media3.0-0v5 - wxWidgets Cross-platform C++ GUI toolkit (GTK+ media library runtime)
libwxgtk-media3.0-dev - wxWidgets Cross-platform C++ GUI toolkit (GTK+ media library development)
libwxgtk-media3.0-gtk3-0v5 - wxWidgets Cross-platform C++ GUI toolkit (GTK+ 3 media library runtime)
libwxgtk-media3.0-gtk3-dev - wxWidgets Cross-platform C++ GUI toolkit (GTK+ 3 media library development)
libwxgtk-webview3.0-gtk3-0v5 - wxWidgets Cross-platform C++ GUI toolkit (GTK+ 3 webview library runtime)
libwxgtk-webview3.0-gtk3-dev - wxWidgets Cross-platform C++ GUI toolkit (GTK+ 3 webview library development)
libwxgtk3.0-0v5 - wxWidgets Cross-platform C++ GUI toolkit (GTK+ runtime)
libwxgtk3.0-dev - wxWidgets Cross-platform C++ GUI toolkit (GTK+ development)
libwxgtk3.0-gtk3-0v5 - wxWidgets Cross-platform C++ GUI toolkit (GTK+ 3 runtime)
libwxgtk3.0-gtk3-dev - wxWidgets Cross-platform C++ GUI toolkit (GTK+ 3 development)
python-wxgtk-media3.0 - Python interface to the wxWidgets Cross-platform C++ GUI toolkit (wx.media)
python-wxgtk-media4.0 - Python interface to the wxWidgets Cross-platform C++ GUI toolkit (wx.media)
python-wxgtk-webview3.0 - Python interface to the wxWidgets Cross-platform C++ GUI toolkit (wx.html2)
python-wxgtk-webview4.0 - Python interface to the wxWidgets Cross-platform C++ GUI toolkit (wx.html2)
python-wxgtk3.0 - Python interface to the wxWidgets Cross-platform C++ GUI toolkit
python-wxgtk3.0-dev - Development files for wxPython
python-wxgtk4.0 - Python interface to the wxWidgets Cross-platform C++ GUI toolkit
python3-wxgtk-media4.0 - Python 3 interface to the wxWidgets Cross-platform C++ GUI toolkit (wx.media)
python3-wxgtk-webview4.0 - Python 3 interface to the wxWidgets Cross-platform C++ GUI toolkit (wx.html2)
python3-wxgtk4.0 - Python 3 interface to the wxWidgets Cross-platform C++ GUI toolkit
```

See Also
--------

- [wxWidgets website](https://www.wxwidgets.org/docs/)
- [wxWidgets source code](https://github.com/wxWidgets/wxWidgets)
- [Python bindings](https://github.com/wxWidgets/wxPython) (source code)

**Comment by cool-RR** [comment]
Sounds good to me. We might as well upgrade to the latest wxPython while
we're at it.

**Comment by bittner** [comment]
Fixing the code quality as mentioned in https://github.com/cool-RR/PythonTurtle/pull/100#issuecomment-421719032 is no fun. This is very unfortunate, because without hard linting QA gate _and_ without tests it will be much more work (and risk!) to migrate the code base.

I want to try it anyway, and the way I do it is to make PythonTurtle run on Python 3, and verify that it continues to run with Python 2. As I mentioned elsewhere, maybe you have an idea about which tests we could add to verify PythonTurtle remains functional.

In the meantime, you could help me by reviewing and merging  #104 and then #105 (I'll probably need to rebase the PR after #104 is merged). I'll then work off the merged result for [migrating the code base to Phoenix](https://docs.wxpython.org/MigrationGuide.html).

**Comment by bittner** [comment]
Fixed with #109.

**Comment by bittner** [comment]
"[Classic vs Phoenix](https://docs.wxpython.org/classic_vs_phoenix.html)" (docs.wxpython.org) was my main resource that helped me transitioning from wxPython 3.0 to 4.0.

Note that I also stripped out all unicode transformations (e.g. from interpreting filenames) as this is [documented to be done by wxPython 4.0](https://docs.wxpython.org/MigrationGuide.html#unicode-and-auto-converting-strings) transparently. The results of this should be verified though on the various target platforms.
